frontend/src/api/open-hands.ts

```diff
@@ -183,18 +196,63 @@ class OpenHands {
    * Get the VSCode URL
    * @returns VSCode URL
    */
-  static async getVSCodeUrl(): Promise<GetVSCodeUrlResponse> {
-    const { data } =
-      await openHands.get<GetVSCodeUrlResponse>("/api/vscode-url");
+  static async getVSCodeUrl(convoId: string): Promise<GetVSCodeUrlResponse> {
+    const { data } = await openHands.get<GetVSCodeUrlResponse>(
+      `/api/conversation/${convoId}/vscode-url`,
+    );
     return data;
   }
 
-  static async getRuntimeId(): Promise<{ runtime_id: string }> {
+  static async getRuntimeId(convoId: string): Promise<{ runtime_id: string }> {
     const { data } = await openHands.get<{ runtime_id: string }>(
-      "/api/conversation",
+      `/api/conversation/${convoId}/config`,
     );
     return data;
   }
+
+  static async searchEvents(
+    conversationId: string,
+    params: {
+      query?: string;
+      startId?: number;
+      limit?: number;
+      eventType?: string;
+      source?: string;
+      startDate?: string;
+      endDate?: string;
+    },
+  ): Promise<{ events: Record<string, unknown>[]; has_more: boolean }> {
+    const { data } = await openHands.get<{
+      events: Record<string, unknown>[];
+      has_more: boolean;
+    }>(`/api/conversation/${conversationId}/events/search`, {
```

Comment:
Maybe we should also use plural nouns for our resource collections (e.g., `/api/conversations/...`). Otherwise it would be best to change `.../events/...` for consistency (though IMO we should use the plural form). WDYT?

Reply:
Good point! Changed. Hopefully I didn't break anything 🙈